src/api/helpers.py

```diff
@@ -135,6 +143,16 @@ def create_page(
     return page_output
 
 
+def validate_year_month(year_month: str) -> str:
+    year, month = map(int, year_month.split("-"))
+    if year > date.today().year:
+        raise ValueError("Input year cannot exceed current year.")
+    if not (1 <= month <= 12):
```

Comment:
another way is to try to construct a date object, catching and re-raising the exception:

```
try:
    year, month = map(int, year_month.split("-"))
    date(year, month, 1)
except ValueError:
    raise ValueError("Invalid year-month format or values. Use 'YYYY-MM'.")
```

I repeat ValueError here to add my own error message